### ISSUE TITLE
Select components from shapefile: Check that fields from shapefile and layer are of the same type

### ISSUE THREAD

**Issue by FLO-2DJJ** [FLO-2DSoftware/qgis-flo-2d-plugin#1334]
Check that the fields selected from the shapefiles are of the same type of the layer fields. For example, if the user selects a string to assign to a number, an error may immediately occur or cascade for a later error, like:

![image](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/20424575/b29905cd-8ee7-4d18-9827-32d62d54a34a)

![image](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/20424575/74fb3cf9-02d7-41ac-a29a-5b0dab967267)

**Comment by rpachaly** [comment]
Hi JJ, I agree with you! This check is a very good feature and should be implemented.

**Comment by FLO-2DJJ** [comment]
Solved in pull request #1344.

**Comment by FLO-2DKaren** [comment]
@FLO-2DJJ @rpachaly 
Would it be hard to back this code off?  It's causing more errors than it fixed.

**Comment by FLO-2DJJ** [comment]
Wait. I'm looking at it.

**Comment by FLO-2DJJ** [comment]
> It's causing more errors than it fixed.

@FLO-2DKaren, I do not see the errors with the projects on which I’m conducting the tests. Could you please send me the project that is giving you errors? Thanks.

**Comment by FLO-2DJJ** [comment]
> It's causing more errors than it fixed.

@FLO-2DKaren, I do not see the errors with the projects on which I’m conducting the tests. Could you please send me the project that is giving you errors? Thanks.

**Comment by FLO-2DJJ** [comment]
> I'm trying to build the project so I can share it for testing but I can't get past the dialog because its looking for inlet type = string but inlet type is a integer 0 - 5.

OK. I see the problem.

**Comment by FLO-2DKaren** [comment]
JJ,

Here is a project and video.  I think Robson's smart code that finds the field by name was overridden by your code.

https://flo-2d.sharefile.com/d-s139be3191fca43cd9eb67937604b4e58

**Comment by FLO-2DKaren** [comment]
@FLO-2DJJ 
Let's use this issue to correct the plugin.  I added it to the milestones so the District can see the progress.

**Comment by FLO-2DJJ** [comment]
This is being addressed in pull request #1348.
DO NOT merge and close the pull request yet.

**Comment by FLO-2DKaren** [comment]
OK but maybe don't do a pull request until its ready so that it doesn't get stagnant.

**Comment by FLO-2DJJ** [comment]
> OK but maybe don't do a pull request until its ready so that it doesn't get stagnant.

You are right @FLO-2DKaren. I just wanted to be able merge the pull request to master with a commit that I made to allow all types of data in the _inlet type_ field only, and then continue with other commits. But it seems that, once the pull request is merged, it is also closed.

So, I'm going to merge this #1348, which shows all the fields for the _inlet type_, and continue with the branch that originated it. DO NOT close the branch 

**1334_allow_all_types_in_lists_in_select_components_from_shapefile** 

where I will make all the next commits before making a final new pull request.

**Comment by FLO-2DJJ** [comment]
Found the problem: after the filtering of types, smart code needs to be changed to find item by substring.
Fixed for inlets. I'll continue with the other components.

**Comment by FLO-2DKaren** [comment]
@FLO-2DJJ I tested this and I think it is working good.  I'm going to close it for now.
